src/Windows/Avalonia.Win32.Interop/Wpf/Direct2DImageSurface.cs

```diff
@@ -148,8 +148,9 @@ public RenderTarget GetOrCreateRenderTarget()
                 return _backBuffer.Target;
 
             if (_image == null || _oldDpi.X != dpi.X || _oldDpi.Y != dpi.Y)
-            {
+            {                
                 _image = new D3DImage(dpi.X, dpi.Y);
+                _oldDpi = dpi;
```

Comment:
Wpf\Direct2DImageSurface.cs(111,24): warning CS0649: Field 'Direct2DImageSurface._oldDpi' is never assigned to, and will always have its default value [C:\GitHub\Avalonia\src\Windows\Avalonia.Win32.Interop\Avalonia.Win32.Interop.csproj]

avoid creating D3DImage in case the resolution is the same as the previous one.

```csharp
            if (_image == null || _oldDpi.X != dpi.X || _oldDpi.Y != dpi.Y)
            {
                _image = new D3DImage(dpi.X, dpi.Y);
                _oldDpi = dpi;
            }
```